### ISSUE TITLE
AWS::SNS::Subscription Cloudformation support

### ISSUE THREAD

**Issue by andreas-antoniou-cko** [localstack/localstack#1191]
Hi

I have issues creating sns subscriptions through Cloudformation. If I attempt to include a `AWS::SNS::Subscription` resource in a Cloudformation template and run `create-stack` then I get a 500 Internal Server error. The full template:

```
AWSTemplateFormatVersion: '2010-09-09'
Description: 'Test'
Resources:
  MySnsTopic:
    Type: AWS::SNS::Topic
    Properties:
      TopicName: MySnsTopic
  MySnsTopicSubscription:
    Type: AWS::SNS::Subscription
    Properties:
      Protocol: sqs
      TopicArn: !Ref MySnsTopic
      Endpoint: !GetAtt
        - MySqsQueue
        - QueueArn
  MySqsQueue:
    Type: AWS::SQS::Queue
    Properties:
      QueueName: MySqsQueue
```

If I replace the references with hard-coded arns like this:

```
TopicArn: arn:aws:sns:eu-west-2:123456789012:MySnsTopic
Endpoint: arn:aws:sqs:elasticmq:000000000000:MySqsQueue
```

Then I get no errors but I get a warning:

`No moto Cloudformation support for AWS::SNS::Subscription`.

I can get around this issue by creating the subscription when creating the topic but the issue there is that I cannot add any attributes and in my case I need `RawMessageDelivery` set to true. So my questions are:

1. Is there no support to create `AWS::SNS::Subscription` resources through Cloudformation or am I doing something wrong?
2. If support is not there can I request it as a new feature?
3. Any suggestions on how I can work around this? I know I could do this through a script at startup but would like to not have any additional script steps other than deploying the Cloudformation stack.

**Comment by leeDav** [comment]
I am also having this issue, but with the [serverless localstack plugin](https://github.com/localstack/serverless-localstack).

Is there an update on this?

Weirdly I can create the subscriptions locally (using CLI through `awslocal`) but the plugin doesn't seem to create the subscriptions on deploy. Works when deployed to AWS, though.

**Comment by russ-dav** [comment]
Is there any update on this issue?

**Comment by andreas-antoniou-cko** [comment]
I looked more into it and the limitation is with [moto ](https://github.com/spulec/moto)- not localstack. I never got around raising this in their repo though. Feel free to do so.

**Comment by whummer** [comment]
Thanks for reporting @andreas-antoniou-cko @leeDav @russ-dav . This should be fixed in #1662 - can you please give it a try with the `latest` version of the Docker image? Closing this issue - please re-open if the problem persists.

**Comment by andreas-antoniou-cko** [comment]
Thanks for looking into this @whummer . I just tested and can confirm that if `TopicArn` and `Endpoint` are hard-coded then it works fine. Using the original snippet still returns the 500 Internal Server Error.
